sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -181,3 +181,36 @@ def test_ipy_notebook():
 
         f.flush()
         assert_equal(json.load(f), example_nb.work_notebook)
+
+
+def test_save_figures():
+    """Test file naming when saving figures. Requires mayavi."""
+    import matplotlib.pylab as plt
+    try:
+        from mayavi import mlab
+    except ImportError:
+        return  # Skip if mayavi is not available.
```

Comment:
The best is to import nose at the top of the file and use `raise nose.SkipTest('Mayavi not installed')`